### PR TITLE
Update routing for template, job, credential pages

### PR DESCRIPTION
**Pull request by keithjgrant**
Updates Template page, Job page, and Credential page to use RoutedTabs. Also updates these routes for more consistent urls (e.g. `/credentials/:id/details` rather than `/credentials/details/:id`)

### CODE REVIEW

**Review by vidyanambiar — APPROVED** [review]
looks great!

**Comment by vidyanambiar** [comment]
There seem to be a couple of test failures due to tabs now being represented as links instead of buttons..

File: cypress/support/component.tsx [hunk 1]
```diff
@@ -58,24 +72,28 @@ Cypress.Commands.add('mount', (component, options, rerenderKey) => {
...
-              <Route path="/:id" element={component} />
+              <Route path={`${route?.path || '/:id'}`} element={component} />
             </Routes>
           </Page>
         </ActiveUserProvider>
       </PageFramework>
     </MemoryRouter>,
-    options,
-    rerenderKey
+    {},
```
Comment (kialam) [inline cypress/support/component.tsx:86]: This is hard to grok/parse. Can you add a comment to clarify why the mount method needs these extra parameters? Somewhat related; can these params be optional?

**Review by kialam — CHANGES_REQUESTED** [review]

File: cypress/support/component.tsx [hunk 2]
```diff
@@ -58,24 +72,28 @@ Cypress.Commands.add('mount', (component, options, rerenderKey) => {
...
+    undefined
```
Comment (kialam) [inline cypress/support/component.tsx:87]: Same concern as above

File: cypress/support/component.tsx (hunk 1 above)
Comment (keithjgrant) [reply cypress/support/component.tsx]: good call. These are the default values there, so they should be okay to omit entirely

**Review by kialam — APPROVED** [review]
👍

**Pull request merged**